state/azure/blobstorage/blobstorage.go

```diff
@@ -91,18 +92,24 @@ func (r *StateStore) Init(metadata state.Metadata) error {
 	p := azblob.NewPipeline(credential, options)
 
 	var containerURL azblob.ContainerURL
+	var URL *url.URL
 	customEndpoint, ok := metadata.Properties[endpointKey]
 	if ok && customEndpoint != "" {
-		URL, parseErr := url.Parse(fmt.Sprintf("%s/%s/%s", customEndpoint, meta.accountName, meta.containerName))
-		if parseErr != nil {
-			return parseErr
+		URL, err = url.Parse(fmt.Sprintf("%s/%s/%s", customEndpoint, meta.accountName, meta.containerName))
+		if err != nil {
+			return err
 		}
 		containerURL = azblob.NewContainerURL(*URL, p)
 	} else {
 		URL, _ := url.Parse(fmt.Sprintf("https://%s.blob.%s/%s", meta.accountName, env.StorageEndpointSuffix, meta.containerName))
```

Comment:
If define URL at L95, need to change this line here or revert related changes.

Reply:
Updated